syllabus/core/curriculum_base.py

```diff
@@ -173,13 +181,45 @@ def _sample_distribution(self) -> List[float]:
         """
         raise NotImplementedError
 
-    def _should_use_startup_sampling(self) -> bool:
-        return self.random_start_tasks > 0 and self.completed_tasks < self.random_start_tasks
+    def _should_use_startup_sampling(self) -> bool:  
+        return self.warmup_strategy != "none" and self.sampled_tasks < self.warmup_tasks
+    
+    def _startup_sample(self, k: int) -> List:
+        sampled_tasks = []
+
+        if isinstance(self.task_space.gym_space, Box):
```

Comment:
I think there should be different behavior for box spaces for "fixed" or "random" sampling. This looks like a good start to a "fixed" implementation, though you'll need to keep track of the index in the grid and sample using that. Should be a relatively small change.

For "random" sampling, I think you can literally just do `self.task_space.gym_space.sample()`